lib/log_file.c

```diff
@@ -60,20 +60,45 @@ _file_close(int32_t t)
 	FILE *f = qb_log_target_user_data_get(t);
 
 	if (f) {
-		fclose(f);
 		(void)qb_log_target_user_data_set(t, NULL);
+		fclose(f);
 	}
 }
 
-static void
-_file_reload(int32_t target)
+static int
+_do_file_reload(const char *filename, int32_t target)
 {
 	struct qb_log_target *t = qb_log_target_get(target);
+	FILE *oldfile = qb_log_target_user_data_get(target);
+	FILE *newfile;
+
+	if (filename == NULL) {
+		filename = t->filename;
+	}
+	newfile = fopen(filename, "a+");
 
-	if (t->instance) {
-		fclose(t->instance);
+	/* Only close oldfile if newfile succeeds */
+	if (oldfile && newfile) {
+		fclose(oldfile);
```

Comment:
I'd write this as `if (oldfile) { fclose(oldfile); }` inside the `if (newfile) { }` block below for clarity. YMMV.